### PR TITLE
docs: decision record about the release process

### PR DESCRIPTION
**Pull request by paullatzelsperger**
This PR contains a decision record about the adaptations needed to make the product-edc build more repeatable and in line with established practices.

### CODE REVIEW

File: docs/decision-records/2023-02-09-release-process/README.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  currently released version. Nothing else.
+- feature development happens _in developers' forks only_ to keep the Git reflog of the `origin` clean.
+
+## Rationale
+
+Having releases depend on snapshot versions of upstream projects, such as EDC, is inherently dangerous, particularly
+when that dependency has not yet reached a final state and breaking changes are to be expected. Most problems will stem
+from breaking changes, such as Java SPIs, APIs and changes in service contracts.
+
+Up until now, the only way out was cherry-picking, which is extremely cumbersome and error-prone, and requires a
```
Comment (ndr-brt) [inline docs/decision-records/2023-02-09-release-process/README.md:23]: in fact, cherry-picking was only used for special cases like described in the last paragraph of the "Use release versions of EDC in releases", this decision doesn't change that, so I would remove this two lines.

File: docs/decision-records/2023-02-09-release-process/README.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+readability and make a release's history readily apparent.
+
+### Nightly builds
+
+Nightly builds are generated according to a fixed schedule (as suggested
+in [this Jira issue](https://jira.catena-x.net/browse/A1IDSC-408)). Upstream EDC will soon begin to publish nightly
+builds as actual releases (as opposed to: snapshots) to a separate OSSRH-operated repository (see
+[this EDC decision record](https://github.com/eclipse-edc/Connector/tree/main/docs/developer/decision-records/2023-02-10-nightly-builds)).
+
+Unfortunately there is no way to automatically trigger the product-edc build whenever a new EDC nightly is
```
Comment (ndr-brt) [inline docs/decision-records/2023-02-09-release-process/README.md:99]: Dependabot could be used

File: docs/decision-records/2023-02-09-release-process/README.md (hunk 1 above)
Comment (paullatzelsperger) [reply docs/decision-records/2023-02-09-release-process/README.md:23]: yes, but now we'd have alternative options, i.e. upgrading to newer EDC releases. cherrypicking is only necessary if those newer releases introduce breaking changes.

**Comment by ndr-brt** [comment]
@paullatzelsperger seems that the [last commit](https://github.com/catenax-ng/product-edc/pull/782/commits/c641045508fc8a4f62f6ac47e0a07532942601ae) deleted the document

**Comment by paullatzelsperger** [comment]
> @paullatzelsperger seems that the [last commit](https://github.com/catenax-ng/product-edc/pull/782/commits/c641045508fc8a4f62f6ac47e0a07532942601ae) deleted the document

you are correct. I had it in my staging area, so I could re-add it again, but have zero clue as to why that happened :/

**Review by ndr-brt — APPROVED** [review]

File: docs/development/decision-records/2023-02-09-release-process/README.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+Product-edc's branching model is already very close to
+the [GitFlow branching model](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow#:~:text=What%20is%20Gitflow%3F,lived%20branches%20and%20larger%20commits)
+which is good. The following changes need to be made:
+
+- feature development happens in forks only. Thus, `feature/` branches don't exist in the `origin`. This is important
+  when moving to OSS.
+- adhere to GitFlow branch naming conventions, i.e. no `docu`, `tryout_something` branches. Increases readability and
+  clarity, improves tool support.
+
+_Other guidelines w.r.t. the review process, merging etc. will follow in a later DR._
```
Comment (DominikPinsel) [inline docs/development/decision-records/2023-02-09-release-process/README.md:71]: We can add such a section/comment when there is a corresponding decision record. We don't know yet when we will get there, so maybe we can remove it for now.

<sub>Dominik Pinsel <dominik.pinsel@mercedes-benz.com>, Mercedes-Benz Tech Innovation GmbH, [legal info/Impressum](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md) </sub>

File: docs/development/decision-records/2023-02-09-release-process/README.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+1. The actual fix is done on `develop` and can be cherry-picked into the `hotfix/0.3.1` branch. No new commits are
+   made directly in that branch.
+2. The actual fix is done on `develop` and must be manually ported into the `hotfix/0.3.1` branch. One or several new
+   commits are made on `hotfix/0.3.1`. This is needed when cherry-picking is not available due to incompatibilities
+   between `develop` and the hotfix branch due to intermittent changes.
+3. The fix is only relevant for the `0.3.1` hotfix, it is not needed in `develop`. This can happen, when the problem is
+   not present on `develop`, because it was already implicitly fixed, or otherwise doesn't exist.
+
+This might produce many branches, and the first `hotfix` makes the release obsolete, but it will greatly help
+readability and make a release's history readily apparent.
```
Comment (DominikPinsel) [inline docs/development/decision-records/2023-02-09-release-process/README.md:90]: If we document that we use GitFlow, this section should become obsolete. Am I right?

<sub>Dominik Pinsel <dominik.pinsel@mercedes-benz.com>, Mercedes-Benz Tech Innovation GmbH, [legal info/Impressum](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md) </sub>

File: docs/development/decision-records/2023-02-09-release-process/README.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+Unfortunately there is no way to automatically trigger the product-edc build whenever a new EDC nightly is
+created. The most reliable method is to periodically query for the latest EDC nightly, e.g. leveraging GitHub's `dependabot` feature, or using the method highlighted in the aforementioned Jira issue:
+
+```shell
+curl <OSSHR_RELEASES_REPO_URL>/org/eclipse/edc/connector-core/maven-metadata.xml | xmllint --xpath "//metadata/versioning/versions/version[last()]" -
+```
+
+That would return something like to `0.0.1-20230213`. As soon as the version string (here: Feb 13th, 2023) matches the
+current date, we can start the nightly. If the EDC nightly doesn't appear within a set timeframe, we throw an error.
```
Comment (DominikPinsel) [inline docs/development/decision-records/2023-02-09-release-process/README.md:107]: The scope of this repository is not to build a connector for the latest EDC version, but to provide a connector for Catena-X. Normally this includes using **not** the lates EDC version, as long as we are not sure there are no issues with it.

We also don't do this for our other dependencies. Maybe you can elaborate why we should do this.

<sub>Dominik Pinsel <dominik.pinsel@mercedes-benz.com>, Mercedes-Benz Tech Innovation GmbH, [legal info/Impressum](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md) </sub>

File: docs/development/decision-records/2023-02-09-release-process/README.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+## Notes for becoming OpenSource
+
+- All artifacts (docker images, helm charts, Maven artifacts) should be published to well-known and publicly accessible
+  locations such as MavenCentral, DockerHub, etc. The GitHub Packages repository is only accessible to authenticated
+  users.
+- When and if Tractus-X becomes an Eclipse project, we'll have to adopt the Eclipse Foundation's publishing guidelines,
+  which prescribes the use of Jenkins for publishing to MavenCentral and OSSRH.
+- Typically, GitHub Actions should perform all verification tasks, running tests, etc. and Jenkins' only purpose is to
+  publish.
```
Comment (DominikPinsel) [inline docs/development/decision-records/2023-02-09-release-process/README.md:117]: This section does not contain a decision, so maybe we can remove it. After this repository is migrated to Tractus-X this section will no require dedicated decision, as its a basic Eclipse requirement.

<sub>Dominik Pinsel <dominik.pinsel@mercedes-benz.com>, Mercedes-Benz Tech Innovation GmbH, [legal info/Impressum](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md) </sub>

**Review by DominikPinsel — CHANGES_REQUESTED** [review]

**Review by jimmarino — APPROVED** [review]

**Review by tuncaytunc-zf — APPROVED** [review]

File: docs/development/decision-records/2023-02-09-release-process/README.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
+# Release process of product-edc
```
Comment (florianrusch-zf) [inline docs/development/decision-records/2023-02-09-release-process/README.md:1]: shouldn't we already use the term `tractusx-edc`?

File: docs/development/decision-records/2023-02-09-release-process/README.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+## Decision
+
+To improve stability, reproducibility and maintainability of releases, product-edc will undergo the following changes:
+
+- use EDC `SNAPSHOT` builds during development
+- use release versions of EDC in releases. Release branches must not change upstream dependency versions, unless there
+  is a clear
+  and concise reason to do so.
```
Comment (florianrusch-zf) [inline docs/development/decision-records/2023-02-09-release-process/README.md:10]:
```suggestion
  is a clear and concise reason to do so.
```

File: docs/development/decision-records/2023-02-09-release-process/README.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+- feature development happens _in developers' forks only_ to keep the Git reflog of the `origin` clean.
+
+## Rationale
+
+Having releases depend on snapshot versions of upstream projects, such as EDC, is inherently dangerous, particularly
+when that dependency has not yet reached a final state and breaking changes are to be expected. Most problems will stem
+from breaking changes, such as Java SPIs, APIs and changes in service contracts.
+
+Up until now, the only way out was cherry-picking, which is extremely cumbersome and error-prone, and requires a
+parallel build pipeline to publish the cherry-picked artifacts of EDC (and potentially others). With the approach presented here, cherry-picking is still an option, but there are easier alternatives to it.
```
Comment (florianrusch-zf) [inline docs/development/decision-records/2023-02-09-release-process/README.md:24]:
```suggestion
parallel build pipeline to publish the cherry-picked artifacts of EDC (and potentially others). With the approach
presented here, cherry-picking is still an option, but there are easier alternatives to it. 
```

**Review by florianrusch-zf — CHANGES_REQUESTED** [review]

**Review by florianrusch-zf — CHANGES_REQUESTED** [review]
Two points:
1. There are still comments from @DominikPinsel open which are in my eyes valid to discuss.
2. The text still uses the wording `product-edc` in some places. @stefan-ettl can you confirm that we need to replace it with tractusx-edc?

**Comment by paullatzelsperger** [comment]
> Two points:
> 
> 1. There are still comments from @DominikPinsel open which are in my eyes valid to discuss.

Most questions that are asked are either beyond the scope of this D-R (e.g. why forking is necessary in Eclipse), are basic knowledge of the Maven release process (e.g. what exactly a `-SNAPSHOT` version is), are based on a misunderstanding (cf. the first comment about the scope being much larger) or are simply not actionable. 
And to be clear, the general content of this decision record was agreed on during the on-site meeting on Feb 8 2023.

In addition, there are already 3 approvals on this PR, which - going by majority vote - would already warrant a merge.

> 2. The text still uses the wording `product-edc` in some places. @stefan-ettl can you confirm that we need to replace it with `tractusx-edc`?

I think replacing `product-edc` should be done in one fell swoop and not piece-by-piece. as far  as it concerns this PR I've replaced all occurrences of "product-edc" with "tractusx-edc".

**Pull request closed without merging**